src/audio/drc/drc_math_lib/asin/c_fixpt/drc_asin.c

```diff
@@ -0,0 +1,221 @@
+/*
+ * File: drc_asin.c
+ *
+ * Author :Shriram Shastry
+ * Copyright (c) 2016, Intel Corporation All rights reserved.
+ */
+
+/* Include Files */
+#include "drc_asin.h"
+
+/* Function Declarations */
+static int asr_s32(int u, unsigned int n);
+static signed char iScalarCordicAsin(signed char sinValue, short nIters, const
+  signed char LUT[6]);
+
+/* Function Definitions */
+
+/*
+ * Arguments    : int u
+ *                unsigned int n
+ * Return Type  : int
+ */
+static int asr_s32(int u, unsigned int n)
+{
+  int y;
+  if (u >= 0)
+  {
+    y = (int)((unsigned int)(((unsigned int)u) >> n));
+  }
+  else
+  {
+    y = (-((int)((unsigned int)(((unsigned int)((int)(-1 - u))) >> n)))) - 1;
```

Comment:
indentation - please use tabs (tab size is 8). Also please use std types like uint32_t or int16_t as we want this code to run exactly the same on 64bit x86  targets.